src/Core/src/Platform/Android/ImageViewExtensions.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Android.Graphics.Drawables;
+using Android.Widget;
+
+namespace Microsoft.Maui
+{
+	public static class ImageViewExtensions
+	{
+		public static void Clear(this ImageView imageView)
+		{
+			// stop the animation
+			if (imageView.Drawable is IAnimatable animatable)
+				animatable.Stop();
+
+			// clear the view and release any bitmaps
+			imageView.SetImageResource(global::Android.Resource.Color.Transparent);
+		}
+
+		public static void UpdateAspect(this ImageView imageView, IImage image)
+		{
+			imageView.SetScaleType(image.Aspect.ToScaleType());
+		}
+
+		public static void UpdateIsAnimationPlaying(this ImageView imageView, IImageSourcePart image)
+		{
+			if (imageView.Drawable is IAnimatable animatable)
+			{
+				if (image.IsAnimationPlaying)
+				{
+					if (!animatable.IsRunning)
+						animatable.Start();
+				}
+				else
+				{
+					if (animatable.IsRunning)
+						animatable.Stop();
+				}
+			}
+		}
+
+		public static async Task<IDisposable?> UpdateSourceAsync(this ImageView imageView, IImageSourcePart image, IImageSourceServiceProvider services, CancellationToken cancellationToken = default)
+		{
+			imageView.Clear();
+
+			image.UpdateIsLoading(false);
+
+			var context = imageView.Context;
+			if (context == null)
+				return null;
+
+			var imageSource = image.Source;
+			if (imageSource == null)
+				return null;
+
+			var events = image as IImageSourcePartEvents;
+
+			events?.LoadingStarted();
+			image.UpdateIsLoading(true);
+
+			try
+			{
+				var service = services.GetRequiredImageSourceService(imageSource);
+				if (service is IApplyImageSourceService applyService)
+				{
+					// use the faster/better way
+
+					var applied = await applyService.ApplyDrawableAsync(imageSource, image, imageView, cancellationToken);
+
+					events?.LoadingCompleted(applied);
+				}
+				else
+				{
+					// fall back to setting it manually
+
+					var result = await service.GetDrawableAsync(imageSource, context, cancellationToken);
+					var drawable = result?.Value;
+
+					var applied = !cancellationToken.IsCancellationRequested && imageSource == image.Source;
+
+					// only set the image if we are still on the same one
+					if (applied)
+					{
+						imageView.SetImageDrawable(drawable);
```

Comment:
should we still validate that the imageView is valid for setting the image? Inside Compatibility we would check if it was destroyed. If the viewmapper can cancel the token that should also help with avoiding this call if it's no longer in a valid state

Reply:
I updated the code to now check if the view was disposed. I already had a check to see if the token was cancelled.